### DISCUSSION TITLE
When using chezmoi add --follow,  chezmoi gets confused due to filemodes

### DISCUSSION THREAD

**Discussion by jglotzer** [twpayne/chezmoi#4091] [General]
Let's say my .bashrc is really a symlink to ~/mydotfiles/.bashrc. The permissions of the symlink itself are 777 but the permissions of the target file are 644. 

If I do something like

```chezmoi init
 chezmoi add --follow ~/.bashrc
 chezmoi cd
 git commit -m "Initial commit"
 git push origin mainline 
```

I end up in a state such that git sees the repo as "clean" (no changes, no diffs) but

```
 chezmoi status
 MM .bashrc
```

and

```
chezmoi diff .bashrc
```
shows filemode diffs ie something like

```
diff --git a/.bashrc b/.bashrc
deleted file mode 120777
index f51c8693f122be8fed58626b180194d3b3de4450..0000000000000000000000000000000000000000
--- a/.bashrc
+++ /dev/null
@@ -1 +0,0 @@
-mydotfiles/.bashrc
diff --git a/.bashrc b/.bashrc
old mode 120777
new mode 100644
index f51c8693f122be8fed58626b180194d3b3de4450..bca7d810b52cb19537480166dcbcca761c2ecef4
--- a/.bashrc
+++ b/.bashrc
```
Followed by the entire file showing up as a diff.

Changing filemode to false (from default which is true) in the chezmoi .git/config doesn't seem to help.

Am I missing some key step? This should be easily reproducible.

Chezmoi built from source, running on Fedora linux

```
chezmoi version v2.54.0, commit efc8ddce480c35fae3c435797dc2114c556baf5c, built at 2024-11-14T00:03:38Z, built by source

```

**Comment by twpayne** [comment]
> Am I missing some key step?

Yes.

`chezmoi diff` is telling you that `.bashrc` is currently a symlink to `mydotfiles/.bashrc` and should be replaced with a regular file with the given contents.

**Reply by jglotzer** [reply]
Got it. Thank you!
